packages/itwinui-react/src/core/ThemeProvider/ThemeProvider.tsx

```diff
@@ -367,9 +378,15 @@ const PortalContainer = React.memo(
 
     if (shouldSetupPortalContainer) {
       return (
-        <div style={{ display: 'contents' }} ref={setPortalContainer} id={id}>
+        <Root
```

Comment:
The goal of this PR is to help with my future PR, where I'm planning to render the portal container as a child of `<body>`, rather than as a child of `<ThemeProvider>`. This will fix many of the popover/stacking-related issues.

Could you elaborate on what kinds of bugs/confusion you're worried about? I don't know why it would be problematic to have two roots, but making this change is a requirement, because the portaled elements will no longer be DOM descendants of the main `Root`. Maybe I can add some tests to ease your concerns.

For some more context: the portal container in iTwinUI v1 used to be a child of `<body>`, but we moved away from that because we want to enable incremental adoption of sub-trees. The reason we were able to do this in v1 is that we were imperatively setting the theme attributes on the `<html>`/`<body>` elements, whereas now we don't do such side-effects.

<details><summary>Old old DOM (simplified)</summary>

```html
<html data-iui-theme="dark">
  <body class="iui-root">
    <button class="iui-button" aria-controls="1">…</button>

    <div>
      <div id="1" class="iui-menu" role="menu">…</div>
    </div>
  </body>
</html>
```

</details>

<details>
<summary>New DOM (simplified)</summary>

```html
<html>
  <body>
    <div class="iui-root" data-iui-theme="dark">
      <button class="iui-button" aria-controls="1">…</button>
    </div>

    <div class="iui-root" data-iui-theme="dark">
      <div id="1" class="iui-menu" role="menu">…</div>
    </div>
  </body>
</html>
```

</details>

Reply:
> Since each ~~DOM~~ theme will be in its own iui-root, I think it'll be possible.

That's right, we can create as many portal roots as we want. I expect we will need one per theme per version.

--

I added an e2e test in [`51610f4`](https://github.com/iTwin/iTwinUI/pull/2173/commits/51610f4667ab08921a176b25db9585fa075c7e46) that verifies that theme is kept in sync between the main root and the portaled root. To make the portal container easy to find in e2e tests, I also added a `data-iui-portal` attribute ([`05b103c`](https://github.com/iTwin/iTwinUI/pull/2173/commits/05b103c96ba43776c24c00c90568e51478ec0f8f)); as a bonus, this will make it easier to debug/disambiguate.